src/main/cpp/events/sampleevent.cpp

```diff
@@ -297,9 +297,21 @@ int SampleEvent::getLoopStartOffset()
     return _loopStartOffset;
 }
 
-void SampleEvent::setLoopStartOffset( int value )
+void SampleEvent::setLoopStartOffset( int value, bool applyCrossfade )
 {
     _loopStartOffset = std::min( value, std::max( 0, _eventLength - 1 ));
+
+    if ( applyCrossfade ) {
+
+        // calculate the amount of samples we deem satisfactory to prevent popping at non-zero crossings
+        int samplesToFade = BufferUtility::millisecondsToBuffer( 2, AudioEngineProps::SAMPLE_RATE );
+        _crossfadeStart = _buffer->bufferSize - ( samplesToFade + 1 ); // at end of sample, prior to looping
+        _crossfadeEnd   = _loopStartOffset + samplesToFade;            // from beginning of loop start offset
+    }
```

Comment:
Updating function to allow fading of samples when the loop is about to occur. This should prevent nasty pops when samples aren't looping at zero crossings.